virtual-desktop/src/app/authentication-manager/login/login.component.ts

```diff
@@ -318,8 +344,27 @@ export class LoginComponent implements OnInit {
               this.expiredPassword = true;
               this.loginMessage = this.translation.translate(PASSWORD_EXPIRED);
             } else {
-              this.errorMessage = this.translation.translate('AuthenticationFailed',
-              { numTypes: failedTypes.length, types: JSON.stringify(failedTypes) });
+              this.errorDetails = '';
+              let err;
+              // Get the server error messages from auth plugins
+              for (let i = 0; i < failedTypes.length; i++) {
```

Comment:
This looks like code duplication from above. If possible, move the logic into a utility function to prevent bugs in the case that we enhance 1 but not the other.